xxHash32/xxHash32.go

```diff
@@ -79,35 +82,34 @@ func (xxh *xxHash) Write(input []byte) (int, error) {
 		xxh.bufused += len(input) - r
 
 		// fast rotl(13)
-		p32 := xxh.v1 + (uint32(xxh.buf[p+3])<<24|uint32(xxh.buf[p+2])<<16|uint32(xxh.buf[p+1])<<8|uint32(xxh.buf[p]))*prime32_2
+		p32 := xxh.v1 + u32(xxh.buf[:])*prime32_2
```

Comment:
go compiler does many interesting optimizations here:
- u32 function is inlined
- on little endian platforms uint32 is read directly from bytes
- bounds checking elimination is propagated inside u32
(tested with go 1.8 on an amd64 platform)